arrangor/src/main/kotlin/no/nav/amt/tiltak/arrangor/ArrangorServiceImpl.kt

```diff
@@ -46,8 +49,33 @@ class ArrangorServiceImpl(
 		return arrangorRepository.getByOrganisasjonsnummer(virksomhetsnummer)?.toArrangor()
 	}
 
-	override fun getVirksomheterForAnsatt(ansattId: UUID): List<Arrangor> {
-		throw NotImplementedError("getVirksomheterForAnsatt in ArrangorService is not yet implemented")
+	override fun oppdaterArrangor(arrangorUpdate: ArrangorUpdate) {
+		val original =  getArrangorByVirksomhetsnummer(arrangorUpdate.organisasjonsnummer) ?:
+			throw NoSuchElementException("Fant ikke Arrangør med virksomhetsnummer ${arrangorUpdate.organisasjonsnummer}")
+
+		val overordnetEnhet = hentOverordnetEnhet(arrangorUpdate, original)
+
+		transactionTemplate.executeWithoutResult {
+			arrangorRepository.update(
+				ArrangorUpdateDbo(
+					id = original.id,
+					navn = arrangorUpdate.navn,
+					overordnetEnhetNavn = overordnetEnhet.navn,
+					overordnetEnhetOrganisasjonsnummer = overordnetEnhet.organisasjonsnummer,
+				)
+			)
+			arrangorRepository.updateUnderenheter(arrangorUpdate.organisasjonsnummer, arrangorUpdate.navn)
+		}
+	}
+
+	private fun hentOverordnetEnhet(arrangorUpdate: ArrangorUpdate, original: Arrangor) : OverordnetEnhet {
+		if (arrangorUpdate.overordnetEnhetOrganisasjonsnummer != original.overordnetEnhetOrganisasjonsnummer) {
+			val nyOverordnetEnhet = arrangorUpdate.overordnetEnhetOrganisasjonsnummer?.let {
+				enhetsregisterClient.hentVirksomhet(it)
+			}
+			return OverordnetEnhet(nyOverordnetEnhet?.navn, nyOverordnetEnhet?.organisasjonsnummer)
```

Comment:
Isteden for å potensielt returnere et objekt med bare nullverdier, så hadde kanskje koden blitt mer tydelig om man returnerte en nullable OverordnetEnhet. Som:
`return arrangorUpdate.overordnetEnhetOrganisasjonsnummer?.let {
				val nyOverordnetEnhet = enhetsregisterClient.hentVirksomhet(it)
				return OverordnetEnhet(nyOverordnetEnhet.navn, nyOverordnetEnhet.organisasjonsnummer)
			}`

Nå er det kanskje ikke så tydelig utenfra at OverordnetEnhet egentlig ikke har noen verdier satt